pkg/network/ebpf/c/http.h

```diff
@@ -184,47 +178,51 @@ static __always_inline void http_read_data(struct __sk_buff *skb, skb_info_t *sk
     }
 }
 
-static __always_inline int http_handle_packet(struct __sk_buff *skb, skb_info_t *skb_info) {
+static __always_inline int http_handle_packet(struct __sk_buff *skb, skb_info_t *skb_info, u16 src_port) {
     char buffer[HTTP_BUFFER_SIZE];
     __builtin_memset(&buffer, '\0', sizeof(buffer));
 
     http_packet_t packet_type = HTTP_PACKET_UNKNOWN;
     http_method_t method = HTTP_METHOD_UNKNOWN;
     http_read_data(skb, skb_info, buffer, &packet_type, &method);
+    http_transaction_t *http = NULL;
 
-    if (packet_type == HTTP_REQUEST) {
-        // Ensure the creation of a http_transaction_t entry for tracking this request
-        http_transaction_t new_entry = {};
-        __builtin_memcpy(&new_entry.tup, &skb_info->tup, sizeof(conn_tuple_t));
-        bpf_map_update_elem(&http_in_flight, &skb_info->tup, &new_entry, BPF_NOEXIST);
-    }
+    http_transaction_t new_entry = { 0 };
+    new_entry.owned_by_src_port = src_port;
 
-    http_transaction_t *http = bpf_map_lookup_elem(&http_in_flight, &skb_info->tup);
-    if (http == NULL) {
-        // This happens when we lose the beginning of a HTTP request
-        return 0;
-    }
-
-    if (packet_type == HTTP_REQUEST) {
-        // We intercepted the first segment of the HTTP *request*
-        http_begin_request(http, method, buffer);
-    } else if (packet_type == HTTP_RESPONSE) {
-        // We intercepted the first segment of the HTTP *response*
+    switch(packet_type) {
+    case HTTP_REQUEST:
+        bpf_map_update_elem(&http_in_flight, &skb_info->tup, &new_entry, BPF_NOEXIST);
+        http = bpf_map_lookup_elem(&http_in_flight, &skb_info->tup);
+        if (http == NULL || http->owned_by_src_port != src_port) {
```

Comment:
When would `http->owned_by_src_port != src_port`, since the tuple (which `src_port` is derived from) is used for the key?

Reply:
The tuple we use for lookup is "normalized" and is always (client, source).
When we're are tracing a request to an external service the socket filter will see something like
```
localIP:externalIP GET / HTTP/1.1
externalIP:localIP  HTTP /1.0 200
```

but when you have a localhost the socket filter sees

Reply:
The tuple we use for lookup is "normalized" and is always (client, source).
When we're are tracing a request to an external service the socket filter will see something like

one TCP segment with
```
clientAddr:serverAddr GET / HTTP/1.1
```

and another one with 
```
serverAddr:clientAddr  HTTP /1.0 200
```

but when you have a local request the socket filter sees
```
clientAddr:serverAddr GET / HTTP/1.1
serverAddr:clienAddr GET / HTTP/1.1
serverAddr:clientAddr HTTP /1.0 200
clientAddr:serverAddr HTTP /1.0 200
```
so the this field is there to ensure that we only process each "part" of the request once

Reply:
I understand that part, but I don't understand how this field prevents double processing. Can you describe the flow of this field usage and where it would prevent duplicate work?

Reply:
Ah, so `src_port` is pre-normalization. That was non-obvious. Perhaps that could be made clearer?

Reply:
I'll update the comments!

Reply:
Done